### PR TITLE
Handle latex_use_xindy bool confval with callable default specially

### PR DESCRIPTION
**Pull request by jfbu**
Fixes: #6900

- Bugfix

This is alternative to #6902. It identifies better the root of the problem but still feels more like a temporary workaround than a fully satisfying fix.

For this reason I did not add a test yet.

### CODE REVIEW

File: sphinx/config.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def convert_overrides(self, name, value):
                                  (name, name + '.key=value'))
             elif isinstance(defvalue, list):
                 return value.split(',')
-            elif isinstance(defvalue, int):
+            elif isinstance(defvalue, int) or name == 'latex_use_xindy':
```
Comment (jfbu) [inline sphinx/config.py:225]: This solves the problem but only for this special case.

**Review by jfbu — COMMENTED** [review]
The problem thus is that configuration value whose default is set according to a callable will never undergo during confoverrides an attempt to convert its value to int type and if it comes from command line it thus remains of str type which ultimately will raise an error if the callable return type is not str but bool or something else.

File: sphinx/config.py (hunk 1 above)
Comment (tk0miya) [reply sphinx/config.py:225]: -1: hard coding is not extensible. I suppose there are more boolean settings in another extension. How about the way like #6908 instead?

File: CHANGES [hunk 2]
```diff
@@ -16,6 +16,9 @@ Features added
 Bugs fixed
 ----------
 
+* #6900: Not possible to set :confval:`latex_use_xindy` via ``-D`` option to
+  :program:`sphinx-build`
```
Comment (tk0miya) [inline CHANGES:20]: 2.2.3 should be used severe bug like crashes. So I prefer to use 2.0 branch instead in such case.

**Comment by jfbu** [comment]
Closing this without merge as #6908 starts a better proposal. I only consider indeed this as a workaround to display where problem lies. Thanks for reviewing!

**Pull request closed without merging**